### PR TITLE
#1176 onAttack for lua

### PR DESCRIPTION
**Pull request by ottml**

### CODE REVIEW

**Review by Flow86 — APPROVED** [review]

File: libs/s25main/world/GameWorld.cpp [hunk 1]
```diff
@@ -779,6 +779,11 @@ void GameWorld::Attack(const unsigned char player_attacker, const MapPoint pt, c
     // Send the soldiers to attack
     unsigned curNumSoldiers = 0;
 
+    if(!potentialAttackers.empty() && HasLua())
+    {
+        GetLua().EventAttack(player_attacker, attacked_building->GetPlayer());
+    }
```
Comment (Spikeone) [inline libs/s25main/world/GameWorld.cpp:785]: maybe It'd be cool to have the number of attackers as well, since we have the count later anyway.

Comment (ottml) [reply libs/s25main/world/GameWorld.cpp]: Good improvement. Added additional parameters for the number of attackers.

**Comment by Flamefire** [comment]
@Flow86 What's up with Jenkins?

**Comment by Flow86** [comment]
> @Flow86 What's up with Jenkins?

some of the packages seem to be broken, I'll try to renovate the pipeline (to get a newer cross compiler etc) but this will take a while

**Comment by JarnoNL79** [comment]
In the gameworld.cpp file the lua onAttack event is added to the _GameWorld::Attack_, if I understand the code correctly this is only for land attacks and there's a seperate method / function to do sea attacks, the _GameWorld::AttackViaSea_ the lua event seems to be missing from there. 

I could misinterpret the code, but I think adding the code below after line 836 should do the trick (could be something I missed for the harbour instead of building) 

```
    if(counter > 0 && HasLua())
    {
        GetLua().EventAttack(player_attacker, attacked_building->GetPlayer(), counter );
    }
```

**Comment by ottml** [comment]
> In the gameworld.cpp file the lua onAttack event is added to the _GameWorld::Attack_, if I understand the code correctly this is only for land attacks and there's a seperate method / function to do sea attacks, the _GameWorld::AttackViaSea_ the lua event seems to be missing from there.
> 
> I could misinterpret the code, but I think adding the code below after line 836 should do the trick (could be something I missed for the harbour instead of building)
> 
> ```
>     if(counter > 0 && HasLua())
>     {
>         GetLua().EventAttack(player_attacker, attacked_building->GetPlayer(), counter );
>     }
> ```

Added the missing call to seaAttack too.

**Review by Flamefire — APPROVED** [review]

**Review by Flow86 — CHANGES_REQUESTED** [review]
Please rebase on latest master (and perhaps squash some commits to get the history a bit more clean)

**Review by Flamefire — APPROVED** [review]

**Comment by ottml** [comment]
@Flow86 Can you review and approve and merge if everything is okay. Thanks

**Review by Flow86 — APPROVED** [review]

**Pull request merged**